src/libraries/Common/src/Interop/Windows/NtDll/Interop.RtlInitUnicodeString.cs

```diff
@@ -0,0 +1,14 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class NtDll
+    {
+        [LibraryImport(Libraries.NtDll)]
+        internal static partial int RtlInitUnicodeString(out UNICODE_STRING result, IntPtr s);
```

Comment:
```suggestion
        internal static partial void RtlInitUnicodeString(out UNICODE_STRING result, IntPtr s);
```

https://learn.microsoft.com/windows/win32/api/winternl/nf-winternl-rtlinitunicodestring

Reply:
just curious, what would happen if 'int' was used here -- we would just receive garbage, presumably.

Reply:
and speaking generally, do we have an options for detecting this kind of error? for example, comparing to the database that cswin32 runs off.

Reply:
> just curious, what would happen if 'int' was used here -- we would just receive garbage, presumably.

Yep. For x86 platforms we get the last value of the `eax`/`rax` register.

> do we have an options for detecting this kind of error? for example, comparing to the database that cswin32 runs off.

Not at present. CsWin32 is the best approach for this. However, it is using a technically invalid mechanism for Roslyn source generators. I suppose that has been fixed now, but originally it was invalid.